src/spirc.rs

```diff
@@ -170,6 +172,16 @@ impl<D: SpircDelegate> SpircManager<D> {
             protocol::spirc::MessageType::kMessageTypePause => {
                 self.delegate.pause();
             }
+            protocol::spirc::MessageType::kMessageTypeNext => {
+            	self.index = (self.index + 1) % self.tracks.len() as u32;
+                let track = self.tracks[self.index as usize];
```

Comment:
It's an iTeufel Air Blue and runs the stock firmware. I assume it uses an official embedded spotify library, possibly the android one.

Here's a video in which I drag tracks around, queue and unqueue them: https://youtu.be/iOv0AOHAYKg

"Spotifly" is the device name of librespot running on my PC, and you see that I can't reorder or (un)queue songs anymore if I use librespot instead of the official client on my smartphone. Perhaps a certain capability field in the spirc messages is needed to tell the Spotify app that librespot supports queue manipulation operations?

When I tried to queue two songs, the following protobuf message arrived:

```js
kMessageTypeReplace Nexus 5 xxx 37 0

version: 1
ident: "yyy"
protocol_version: "2.0.0"
seq_nr: 37
typ: kMessageTypeReplace
device_state {
    sw_version: "4.4.0.1008/release-1.30.0.0.g1aaef56"
    is_active: false
    can_play: true
    volume: 65535
    name:"Nexus 5"
    became_active_at: 0
    capabilities {
        typ: kSupportedContexts
        stringValue: "album"
        stringValue: "playlist"
        stringValue: "search"
        stringValue: "inbox"
        stringValue: "toplist"
        stringValue: "starred"
        stringValue: "publishedstarred"
        stringValue: "track"
    }
    capabilities {
        typ: kSupportedTypes
        stringValue: "audio/episode"
        stringValue: "audio/local"
        stringValue: "audio/track"
        stringValue: "local"
        stringValue: "track"}
    capabilities {
        typ: kCanBePlayer
        intValue: 1
    }
    capabilities {
        typ: kRestrictToLocal
        intValue: 0
    }
    capabilities {
        typ: kDeviceType
        intValue: 3
    }
    capabilities {
        typ: kGaiaEqConnectId
        intValue: 1
    }
    capabilities {
        typ: kSupportsLogout
        intValue: 1
    }
    capabilities {
        typ: kIsObservable
        intValue: 1
    }
    capabilities {
        typ: kVolumeSteps
        intValue: 0
    }
}
state {
    context_uri: ""
    position_ms: 4294965356
    status: kPlayStatusPlay
    position_measured_at: 1452810692492
    context_description: ""
    shuffle: false
    repeat: false
    playing_from_fallback: true
    row: 0
    playing_track_index: 9
    track {gid: "\212\250\307u\271\311I\351\250F~p\027\344\241z"}
    track {gid: "\2244\242\255+\273@\263\261\260\337\033\267Uu\375"}
    track {gid: "\224\004zL#\235HF\273/\325\244\230\201\262\237"}
    track {gid: "\013\004\304!V\220@\012\200\266\306\3351~\236\352"}
    track {gid: ":\241\221\351\237\246Ly\262K\004G\265\241_\217"}
    track {gid: "\344I\222\002Q\030J\000\233\3754\322$\"\"\005"}
    track {gid: "di\336\034 B@\257\257\005X\247\304X\017U"}
    track {gid: "X0\370\372OaO\245\227M\037\3179\333X\022"}
    track {gid: "q\361\002\2462\003Ms\227\363\305?k\377\354\212"}
    track {gid: "wp\252NI/E\341\265\0210\005\325\260I\333"}
    track {
        gid: "\367\317\211\001m\253A\221\251\360U\333\253O\250\003"
        queued: true
    }
    track {
        gid: "\3257\235\023J\010G\330\204\373\376*[}\312\031"
        queued: true
    }
    track {gid: "\200\250\260\243\231\230A\"\237\346\345\267X|\333\376"}
    track {gid: "\367\317\211\001m\253A\221\251\360U\333\253O\250\003"}
    track {gid: "\3257\235\023J\010G\330\204\373\376*[}\312\031"}
    track {gid: "\345\310V\205%\333B\273\277\251%\361'G\267\374"}
    track {gid: "\031V\354\250\233\035IQ\263\374p\304)\241f\312"}
    track {gid: "\273i[\227ZTDE\246G?6\376h=\242"}
    track {gid: "\353\370\227g<TB\326\262\335\2678\371\202\301\237"}
    track {gid: "\263\310\002y\012\252O\203\255\266\312\013l<oj"}
    track {gid: "\367\210\243\033:XL\271\210\025K\353\310\246\301\366"}
    track {gid: "'\374eK\312\247J\031\264=\035`\005\013\2275"}
    track {gid: "l\375\025)$\314D\303\214\370P\24000\204\210"}
    track {gid: "\013\211;\362B\331Kk\217\230\023\212\301Kt\362"}
    track {gid: "\341y\320AY\252Ns\227n\265\251\022C\032\206"}
    track {gid: "\235:\235\324*YG\301\234|\226\3044r\276\373"}
    track {gid: "\002\302\264\237\306$K\235\260\366<W\210F<."}
    track {gid: "p\032Z\375\337\307H\017\213G\245\321\236\220\015\030"}
    track {gid: "uL>A\236\015D\365\253\366\314\246\361\314\220\030"}
    track {gid: "B\366\260\207\224YK\222\266Hz\252\"9\355\362"}
    track {gid: "h\366\275\353\317wHF\277wk\364\320\027\252H"}
    track {gid: "\000(\302H\325zG\273\263{\000kr\275\377o"}
    track {gid: "\276\"\265\247\023\361DM\235o\225/\316\334\372\375"}
    track {gid: "\015\262j\275\364WI'\210\214\264\313\000\332\027\352"}
    track {gid: "\313Z\374~\304\366G\307\224xC~og\031<"}
    track {gid: "\"b_\220\030\202Iv\220Y\355T\235\236s\247"}
    track {gid: "9}\212&\344\301D\331\207\241\2266x\205\203}"}
    track {gid: "\322;\237\337WF@\320\210\300\237\241<\253\255\036"}
    track {gid: "$\013;{\274=B\245\232\265'\355_\334\256\377"}
    track {gid: "\216\366$\224,\000E1\274>\250LZ\304\026\372"}
    track {gid: "\360\330\026\2606\335G#\202I:\007\\\376,\313"}
    track {gid: "ntL\253e\353H\376\235\275\226\325\337\311\210\224"}
    track {gid: "|\353o&\310\306@b\232\032\305p\213\324\262\270"}
}
recipient: "zzz"
```

There are only two "queued" tracks in the tracklist, near the currently playing title. Those must be the tracks I queued. It looks like the Spotify app just sent the entire new tracklist to librespot instead of telling it what to replace with what. It probably does the same on re-ordering and deletions.